pypsa/plot.py

```diff
@@ -126,23 +132,31 @@ def plot(network, margin=0.05, ax=None, drawmap=True, bus_colors='b',
         logger.error("Matplotlib is not present, so plotting won't work.")
         return
 
+    if basemap is not None:
+        warnings.warn("`basemap` is deprecated, use `geomap` instead.",
+                      DeprecationWarning)
+        geomap = basemap
+
+
     if ax is None:
-        if cartopy_present and drawmap:
-            ax = plt.gca(projection=ccrs.PlateCarree())
+        if cartopy_present and geomap:
+            if projection is None:
+               projection = get_projection_from_crs(network.srid)
+
+            ax = plt.gca(projection=projection)
         else:
             ax = plt.gca()
     else:
-        if cartopy_present and drawmap:
+        if cartopy_present and geomap:
             import cartopy.mpl.geoaxes
             assert isinstance(ax, cartopy.mpl.geoaxes.GeoAxesSubplot), (
                     'The passed axis is not a GeoAxesSubplot. You can '
                     'create one with: \nimport cartopy.crs as ccrs \n'
                     'fig, ax = plt.subplots('
                     'subplot_kw={"projection":ccrs.PlateCarree()})')
 
-    if drawmap:
-        geomap, x, y = draw_map(network, jitter, ax, boundaries,
-                                margin, drawmap)
+    if geomap:
+        x, y = draw_map(network, jitter, ax, boundaries, margin, geomap)
```

Comment:
I don't think I like that `jitter` does not work if one does not have any map. I think it would be better to pull it back into the main function and hand over `x` and `y`.